lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2084,17 +2095,18 @@ LogicalResult FIRRTLLowering::setLowering(Value orig, Value result) {
            "Lowering didn't turn a FIRRTL value into a non-FIRRTL value");
 
 #ifndef NDEBUG
-    auto srcWidth =
-        origType.cast<FIRRTLType>().getPassiveType().getBitWidthOrSentinel();
-
-    // Caller should pass null value iff this was a zero bit value.
-    if (srcWidth != -1) {
-      if (result)
-        assert((srcWidth != 0) &&
-               "Lowering produced value for zero width source");
-      else
-        assert((srcWidth == 0) &&
-               "Lowering produced null value but source wasn't zero width");
+    if (auto baseType = origType.dyn_cast<FIRRTLBaseType>()) {
```

Comment:
`LowerToHW` should expect to get only `BaseType`, any `RefType` surviving till `LowerToHW` should be an error.